src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
+        [HttpGet("[action]/{locationStub}/{featureStub}")]
+        public IActionResult Locations(string locationStub, string featureStub)
+        {
+            Location[] locations = {  };
+
+
+            Feature[] features = {  };
+
+
+            LocationFeature[] locationsFeat = {  };
+
+            var group = new Group { Id = 1, GroupType="distance" };
+
+
+            LocationGroup[] locagroups = { };
+
+
+            if (string.IsNullOrEmpty(locationStub))
+            {
+
+                return View("Locations",locations);
+            }
+            else if (string.IsNullOrEmpty(featureStub))
+            {
+                var locationViewModel = new LocationViewModel();
+
+                foreach (var location in locations)
+                {
+                    if (location.Stub == locationStub)
+                    {
+                        locationViewModel.Location = location;
+                        var featlist = new List<LocationsFeaturesViewModel>();
+                        var neighbors = new List<Location>();
+                        foreach (var item in locationsFeat)
+                        {
+                            if (item.LocationId == location.Id)
+                            {
+                                foreach (var feat in features)
+                                {
+                                    if (item.FeatureId == feat.Id)
+                                    {
+                                        var locafeat = new LocationsFeaturesViewModel
+                                        {
+                                            Text = item.Text,
+                                            Stub = feat.Stub,
+                                            Name = feat.Name,
+                                            ImagePath = feat.ImagePath,
+                                            FontAwesome = feat.FontAwesome,
+                                            BodyText = feat.BodyText,
+                                            RedirectUrl = feat.Name == "Facebook"
+                                            ? location.Facebook
+                                            : item.RedirectUrl
+                                        };
+                                        featlist.Add(locafeat);
+                                    }
+                                }
+                            }
+                        }
+                        var groupid = locagroups.First(c => c.LocationId == location.Id && c.GroupId == 1).GroupId;
+                        foreach (var groupy in locagroups)
+                        {
+                            if (locations.First(c => c.Id == groupy.LocationId).Id != location.Id)
+                            {
+                                if (groupid == groupy.GroupId)
+                                {
+                                    var obj = locations.First(c => c.Id == groupy.LocationId);
```

Comment:
`obj` is not ideal, this could actually just be `neighbors.Add(locations.First(_ => _.Id == groupy.LocationId))`